api/v1alpha1/ipaddress_webhook.go

```diff
@@ -23,23 +26,31 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
 )
 
-func (c *IPAddress) SetupWebhookWithManager(mgr ctrl.Manager) error {
+func (webhook *IPAddress) SetupWebhookWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewWebhookManagedBy(mgr).
-		For(c).
+		For(webhook).
+		WithDefaulter(webhook, admission.DefaulterRemoveUnknownOrOmitableFields).
+		WithValidator(webhook).
 		Complete()
 }
 
 // +kubebuilder:webhook:verbs=create;update,path=/validate-ipam-metal3-io-v1alpha1-ipaddress,mutating=false,failurePolicy=fail,groups=ipam.metal3.io,resources=ipaddresses,versions=v1alpha1,name=validation.ipaddress.ipam.metal3.io,matchPolicy=Equivalent,sideEffects=None,admissionReviewVersions=v1;v1beta1
 // +kubebuilder:webhook:verbs=create;update,path=/mutate-ipam-metal3-io-v1alpha1-ipaddress,mutating=true,failurePolicy=fail,groups=ipam.metal3.io,resources=ipaddresses,versions=v1alpha1,name=default.ipaddress.ipam.metal3.io,matchPolicy=Equivalent,sideEffects=None,admissionReviewVersions=v1;v1beta1
 
-var _ webhook.Defaulter = &IPAddress{}
-var _ webhook.Validator = &IPAddress{}
+var _ webhook.CustomDefaulter = &IPAddress{}
+var _ webhook.CustomValidator = &IPAddress{}
 
-func (c *IPAddress) Default() {
+func (webhook *IPAddress) Default(_ context.Context, _ runtime.Object) error {
+	return nil
 }
 
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type.
-func (c *IPAddress) ValidateCreate() (admission.Warnings, error) {
+func (webhook *IPAddress) ValidateCreate(_ context.Context, obj runtime.Object) (admission.Warnings, error) {
+	c, ok := obj.(*IPAddress)
```

Comment:
I am a bit confused here, if you are generating `c `from `obj` where is this `webhook *IPAddress` used?

Reply:
webhook *IPAddress is not used in this function. 
We dot need to use it.

Reply:
ow ya, its just a receiver var.

Reply:
`(webhook *IPAddress)` is needed because we are implementing CustomValidator interface:
https://github.com/kubernetes-sigs/controller-runtime/blob/bd9c786e30178e26be5f5a5756dbc120c11f2e0e/pkg/webhook/admission/validator_custom.go#L35